### PR TITLE
Jessie chat page

### PR DESCRIPTION
**Pull request by JessieWang0706**
## Overview

Implemented the chat page, chat home, availability, negotiation, and upload pictures. 

## Changes Made
Four types of chats possibilities are implemented, 
1. Negotiation, normal text
2. Availability
3. Picture
4. Product cards

Chat home page with products, seller name, and recent message

Specifics on availability:
1. New event dotted border, light purple background
2. Complete event, purple background
3. On delete event, white border light purple background
4. Able to combine events when overlapping
5. Continue button reset schedule
Can click into the availability bubble to show a schedule

## Test Coverage
Tested on android API 29, iPhone 11pro max

## Problems
Cannot delete availability in text input at this point. 
Scroller in the availability bubble not smooth enough

## Next Steps (delete if not applicable)
Networking stuff.

## Screenshots (delete if not applicable)

<details>

  <summary>Chat Functionality Screenshot</summary>

https://user-images.githubusercontent.com/28740646/163720853-99b79470-f3aa-415f-abc3-849641464358.MP4

</details>

### CODE REVIEW

**Comment by Haichen-Wang** [comment]
Please add more detailed description & screenshot for your PR

File: components/ChatTabs.tsx [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+            {purchaseUnread !== 0 && (
+              <View
+                style={{
+                  width: 20,
+                  height: 16,
+                  backgroundColor: "#FF0000",
+                  borderRadius: 8,
+                  alignItems: "center",
+                  justifyContent: "center",
+
```
Comment (Haichen-Wang) [inline components/ChatTabs.tsx:83]: empty line

File: components/CustomizedNumKeyBoard.tsx [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+    ) {
+      setInput(input.concat(text));
+    }
+  };
+  const onContinueClicked = (original: string) => {
+    setModalVisible(!modalVisible);
+    if (input != "") {
+      setOriginalText(
+        original.concat(
+          "Hi! I'm interested in buying your Blue Pants, but would you be open to selling it for $" +
```
Comment (Haichen-Wang) [inline components/CustomizedNumKeyBoard.tsx:36]: "Blue Pants" should be the item name and should not be hard coded

File: components/ProductList.tsx [hunk 3]
```diff
@@ -29,15 +29,20 @@ export function ProductList({ count, filter, data, navigation }) {
...
     }
+
     return show ? (
       <ProductCard
         title={item.title}
         price={item.price}
         image={item.image}
-        onPress={() => navigation.navigate("ProductHome")}
+        onPress={() => {
+          navigation.setOptions({ animationEnabled: false });
```
Comment (Haichen-Wang) [inline components/ProductList.tsx:39]: we don't want to disable animation right

File: navigation/index.tsx [hunk 4]
```diff
@@ -103,6 +107,27 @@ function RootNavigator() {
       <Stack.Group screenOptions={{ presentation: "modal" }}>
         <Stack.Screen name="Modal" component={ModalScreen} />
         <Stack.Screen name="NewPost" component={PostScreen} />
+        <Stack.Screen
```
Comment (Haichen-Wang) [inline navigation/index.tsx:110]: as discussed, let's make product details a screen instead. I can take care of it tho bc I have it changed on my end

File: screens/SearchScreen.tsx [hunk 5]
```diff
@@ -0,0 +1,186 @@
+import * as React from "react";
```
Comment (Haichen-Wang) [inline screens/SearchScreen.tsx:1]: isn't this a part of aarushi's PR?

File: screens/SignIn.tsx [hunk 6]
```diff
@@ -1,85 +1,84 @@
...
-            </View>
-            
+  return (
+    <View style={styles.container}>
+      {/* <Image
+        style={styles.gradient0}
+        width={"100%"}
+        height={"70%"}
+        source={require("../assets/images/signinbackgroundhue.png")}
+      /> */}
```
Comment (Haichen-Wang) [inline screens/SignIn.tsx:31]: should we have these uncommented?

**Review by Haichen-Wang — CHANGES_REQUESTED** [review]
This is amazing work Jessie!! Can't believe we got the chat working. I've made some comments but no big issues.

File: screens/SearchScreen.tsx (hunk 5 above)
Comment (JessieWang0706) [reply screens/SearchScreen.tsx:1]: because I rebased on her branch. It would not conflict with the main.

**Review by Haichen-Wang — APPROVED** [review]
Your changes have addressed all my concerns! Great work

**Pull request merged**